### DISCUSSION TITLE
TypeScript throws errors when importing non-existing modules in `package/ui`

### DISCUSSION THREAD

**Discussion by magnusriga** [vercel/turborepo#7420] [Help] [answered]
I have a monorepo (Turborepo) where a Next.js app package is importing various components from a shared package (`package/ui`). That shared package is not compiled, as per the new Turborepo examples, it is simply a package containing various components imported directly by other apps in the monorepo.

The problem is that those shared components have imports like this:

`import Link from 'next/link'`

TypeScript throws an error, because it cannot see packages like `next/link` from within the shared components package (it does not have a node_modules). And I use pnpm, so most external npm packages are only visible to the package that uses them, they are not hoisted to the root of the monorepo.

Is there a good solution, i.e. somehow tell typescript to not throw an error on imports it cannot resolve?

**Comment by anthonyshew** [comment, marked as answer]
The canonical answer to this is to create a component in the UI package that accepts `Link` as a prop. Then, you use that prop in the render of the component. That way, you won't be installing `next` into the UI package but you can still use soft-navigation.
